src/auto-evo/mutation_strategy/CommonMutationFunctions.cs

```diff
@@ -0,0 +1,192 @@
+
+        position = GetRealisticPosition(organelle, newSpecies.Organelles, direction, random);
+
+        newSpecies.Organelles.Add(position);
+
+        // If the new species is a eukaryote, mark this as such.
+        if (organelle == Nucleus)
+        {
+            newSpecies.IsBacteria = false;
+        }
+    }
+
+    public static OrganelleTemplate GetRealisticPosition(OrganelleDefinition organelle,
+        OrganelleLayout<OrganelleTemplate> existingOrganelles, Direction direction, Random random)
+    {
+        var result = new OrganelleTemplate(organelle, new Hex(0, 0), 0);
+
+        // Loop through all the organelles and find an open spot to
+        // place our new organelle attached to existing organelles
+        // This almost always is over at the first iteration, so its
+        // not a huge performance hog
+        foreach (var otherOrganelle in existingOrganelles.OrderBy(_ => random.Next()))
+        {
+            // The otherOrganelle is the organelle we wish to be next to
+            // Loop its hexes and check positions next to them
+            foreach (var hex in otherOrganelle.RotatedHexes)
+            {
+                // Offset by hexes in organelle we are looking at
+                var pos = otherOrganelle.Position + hex;
+
+                foreach (int side in SideTraversalOrder(hex, direction, random))
+                {
+                    // pick a hex direction, with a slight bias towards forwards
+                    for (int radius = 1; radius <= 3; ++radius)
+                    {
+                        // Offset by hex offset multiplied by a factor to check for greater range
+                        var hexOffset = Hex.HexNeighbourOffset[(Hex.HexSide)side];
+                        hexOffset *= radius;
+                        result.Position = pos + hexOffset;
+
+                        if (organelle.HasMovementComponent)
+                        {
+                            // Face movement to move forward
+                            result.Orientation = 3;
+
+                            if (existingOrganelles.CanPlace(result, new List<Hex>(), new List<Hex>()))
+                            {
+                                return result;
+                            }
+                        }
+
+                        // Check every possible rotation value.
+                        for (int rotation = 0; rotation <= 5; ++rotation)
+                        {
+                            result.Orientation = rotation;
+
+                            if (existingOrganelles.CanPlace(result, new List<Hex>(), new List<Hex>()))
+                            {
+                                return result;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // We didn't find an open spot, this doesn't make much sense
+        throw new Exception("Mutation code could not find a good position " +
```

Comment:
Exceptions for kind of foreseeable things is not good, I tracked down one problem with the game completely freezing in the .NET runtime to throwing background thread exceptions. I modified the copy in master of finding a good organelle position to return a nullable value I think. Maybe this copy of the function hasn't gotten the latest fixes?